File: c/Colleen.c
/*
	Created by Hugo MARTZOLF on 3/4/19.
*/
#include <stdio.h>
void Colleen(const char *s){printf(s,10,9,34,s);}int main(){
/*
	This program will print its own source when run.
*/
Colleen("/*%1$c%2$cCreated by Hugo MARTZOLF on 3/4/19.%1$c*/%1$c#include <stdio.h>%1$cvoid Colleen(const char *s){printf(s,10,9,34,s);}int main(){%1$c/*%1$c%2$cThis program will print its own source when run.%1$c*/%1$cColleen(%3$c%4$s%3$c);return 0;}%1$c");return 0;}

File: rust/Colleen.rs
macro_rules!c{()=>{"macro_rules!c{{()=>{{{1}{2}{1}}}}}{0}/*{0}    Created by Hugo MARTZOLF on 24/11/19.{0}*/{0}fn printer (){0}{{{0}    println!(c!(), 10 as char, 34 as char, c!()){0}}}{0}fn main(){0}{{{0}/*{0}    mandatory comment{0}*/{0}    printer(){0}}}"}}
/*
    Created by Hugo MARTZOLF on 24/11/19.
*/
fn printer ()
{
    println!(c!(), 10 as char, 34 as char, c!())
}
fn main()
{
/*
    mandatory comment
*/
    printer()
}
